Library/Homebrew/cmd/update-report.rb

```diff
@@ -126,131 +127,137 @@ def output_update_report
+            DescriptionCacheStore.new(db)
+                                 .update_from_report!(hub)
+          end
+          CacheStoreDatabase.use(:cask_descriptions) do |db|
+            CaskDescriptionCacheStore.new(db)
+                                     .update_from_report!(hub)
+          end
         end
+        puts if args.auto_update?
+      elsif !args.auto_update? && !ENV["HOMEBREW_UPDATE_FAILED"] && !ENV["HOMEBREW_MIGRATE_LINUXBREW_FORMULAE"]
+        puts "Already up-to-date." unless args.quiet?
       end
-      puts if args.auto_update?
-    elsif !args.auto_update? && !ENV["HOMEBREW_UPDATE_FAILED"] && !ENV["HOMEBREW_MIGRATE_LINUXBREW_FORMULAE"]
-      puts "Already up-to-date." unless args.quiet?
-    end
 
-    Commands.rebuild_commands_completion_list
-    link_completions_manpages_and_docs
-    Tap.each(&:link_completions_and_manpages)
+      Commands.rebuild_commands_completion_list
+      link_completions_manpages_and_docs
+      Tap.each(&:link_completions_and_manpages)
 
-    failed_fetch_dirs = ENV["HOMEBREW_MISSING_REMOTE_REF_DIRS"]&.split("\n")
-    if failed_fetch_dirs.present?
-      failed_fetch_taps = failed_fetch_dirs.map { |dir| Tap.from_path(dir) }
+      failed_fetch_dirs = ENV["HOMEBREW_MISSING_REMOTE_REF_DIRS"]&.split("\n")
+      if failed_fetch_dirs.present?
+        failed_fetch_taps = failed_fetch_dirs.map { |dir| Tap.from_path(dir) }
 
-      ofail <<~EOS
-        Some taps failed to update!
-        The following taps can not read their remote branches:
-          #{failed_fetch_taps.join("\n  ")}
-        This is happening because the remote branch was renamed or deleted.
-        Reset taps to point to the correct remote branches by running `brew tap --repair`
-      EOS
+        ofail <<~EOS
+          Some taps failed to update!
+          The following taps can not read their remote branches:
+            #{failed_fetch_taps.join("\n  ")}
+          This is happening because the remote branch was renamed or deleted.
+          Reset taps to point to the correct remote branches by running `brew tap --repair`
+        EOS
+      end
     end
 
     return if new_repository_version.blank?
 
     puts
     ohai "Homebrew was updated to version #{new_repository_version}"
-    Settings.write "latesttag", new_repository_version
-    if new_repository_version.split(".").last == "0"
-      puts <<~EOS
-        More detailed release notes are available on the Homebrew Blog:
-          #{Formatter.url("https://brew.sh/blog/#{new_repository_version}")}
-      EOS
-    elsif !args.quiet?
+    new_major_version, new_minor_version = (new_repository_version.split(".")[0, 2]).map(&:to_i)
+    if old_repository_version.present?
+      old_major_version, old_minor_version = (old_repository_version.split(".")[0, 2]).map(&:to_i)
+    end
+    if old_repository_version.blank? || (new_major_version > old_major_version) \
+        || (new_minor_version > old_minor_version)
```

Comment:
```suggestion
    if !args.quiet? && (old_repository_version.blank? || (new_major_version > old_major_version))
```

It doesn't seem right to output this if only the minor versions differ or `--quiet` has been passed.

Reply:
I am sure if this is an interpretation difference but the blog entries are for me minor releases (major.minor.revision). If you go from version 2.2.1 to 2.3.2 you will need to evaluate the minor difference. If you go from 2.4.1 to version 3.2.2. the major.

I agree with the --quiet addition. I will add it.

Reply:
Ah, sorry, I misread `minor` as `patch` here 🤦🏻